### PR TITLE
Simplify isort configuration using "black" profile

### PR DESCRIPTION
**Pull request by deepyaman**
Signed-off-by: Deepyaman Datta <deepyaman.datta@utexas.edu>

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro/pull/1630"></a>

### CODE REVIEW

**Review by AntonyMilneQB — APPROVED** [review]
This is great, I didn't know about this `profile` option before. Do we still want this line though? I'm not sure what it currently does to be honest.
```
known_third_party = "kedro"
```

Also remember to update the starters ⭐

**Comment by deepyaman** [comment]
> This is great, I didn't know about this `profile` option before.

It's relatively new (as of `isort==5.0`).

> Do we still want this line though? I'm not sure what it currently does to be honest.
> ```
> known_third_party = "kedro"
> ```

No, `isort` will detect it's third-party. You don't list all your other libs as third-party.

> Also remember to update the starters ⭐

🚀

**Comment by deepyaman** [comment]
> Also remember to update the starters ⭐

Done in https://github.com/kedro-org/kedro-starters/pull/97!

File: test_requirements.txt [hunk 1]
```diff
@@ -17,6 +17,7 @@ hdfs>=2.5.8, <3.0
 holoviews~=1.13.0
 import-linter[toml]==1.2.6
 ipython>=7.31.1, <8.0
+isort~=5.0
```
[the remark this answers is not in the document]
Comment (deepyaman) [reply test_requirements.txt:20]: @MerelTheisenQB [`isort` was always required via `pylint`](https://github.com/PyCQA/pylint/blob/d51110e6cd3ad7a44725f9fe915e0ac8779bffe7/setup.cfg#L36); it just so happens that the `profile` option requires a stricter lower bound, hence the addition.

It honestly won't make a difference currently; I checked prior to making this change that a 5.x version of `isort` was getting resolved/installed already.

**Review by MerelTheisenQB — APPROVED** [review]

**Pull request merged**